app/controllers/users.controller.js

```diff
@@ -52,41 +40,26 @@ exports.createAccount = function (req, res) {
                 res.status(400).send({successful: false, text: 'The email address, ' +  user.email
                     + ' is already associated with another account.'});
             } else {
-                const failureMessage = 'Your account was created, however we could not send a verification email.';
+                const errorMessage = 'Your account was created, however we could not send a verification email.';
 
-                var token = new VerificationToken({user: user._id, token: crypto.randomBytes(16).toString('hex')});
+                var token = new VerificationToken({user: user._id});
 
                 token.save(function (err) {
                     if (err) {
-                        res.status(500).send({successful: true, text: failureMessage});
+                        res.status(500).send({successful: true, text: errorMessage});
                     } else {
-                        nodemailer.createTransport(TRANSPORTER).sendMail({
-                            from: EMAIL_FROM,
-                            to: user.email,
-                            subject: EMAIL_VERIFY_SUBJECT,
-                            text: 'Hello ' + user.firstName + ' ' + user.lastName + ',\n\nPlease verify your GT '
-                            + 'ThriftShop account by clicking the link:\nhttp:\/\/' + config.uiUrl + '\/verify/'
-                            + token.token
-                        }, function (err) {
-                            if (err) {
-                                res.status(503).send({successful: true, text: failureMessage});
-                            } else {
-                                res.status(200).send({successful: true, text: 'Check your email for a link to verify '
-                                    + 'your account.'
-                                });
-                            }
-                        });
+                        sendVerificationEmail(user, token, errorMessage, true, res);
                     }
                 });
             }
         });
     }
 };
 
-exports.resendVerification = function (req, res, next) {
+exports.resendVerificationEmail = function (req, res, next) {
     var email = (req.body.email) ? req.body.email.trim().toLowerCase() : '';
 
-    if (!EMAIL_REGEX.test(email)) {
+    if (!EmailUtils.EMAIL_REGEX.test(email)) {
```

Comment:
I'd like to see `EmailUtils.EMAIL_REGEX.test(email)` be a method, say `EmailUtils.testEmail(email)` in the `EmailUtils` class. That'd be better encapsulation, fewer details for the client of `EmailUtils` to worry about and allows change within `EmailUtils` down the road without affecting this controller.

Reply:
Good idea.